errors.go

```diff
@@ -2,6 +2,29 @@ package fargo
 
 // MIT Licensed (see README.md) - Copyright (c) 2013 Hudl <@Hudl>
 
+import (
+	"fmt"
+)
+
+type unsuccessfulHTTPResponse struct {
+	statusCode    int
```

Comment:
I see. If you feel it's more appropriate leave it. I was just feeling that's it might look better when creating new error to have message first, so it would be similar to `printf` function. I'll leave decision to you. Thanks for clarifying.